### PR TITLE
Persist SSL configuration for appgateway

### LINKED ISSUE #287: azure_rm_appgateway ssl_policy settings have no effect
##### SUMMARY

Setting **_ssl_policy_** to either **predefined** or **custom** polices have no effect on the configuration of the gateway.
Changing the configuration manually on the Azure portal and re running the ansible playbook results in the Default ssl_policy  always being applied.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_appgateway
azure.azcollection.azure_rm_appgateway

##### ANSIBLE VERSION

```paste below
ansible 2.10.2
```

##### CONFIGURATION

```paste below
dvnt@D34dB33F:~$ ansible-config dump --only-changed
dvnt@D34dB33F:~$

```

##### OS / ENVIRONMENT

Azure Cloud Shell
Ubuntu 18.04
Ubuntu 20.04

##### STEPS TO REPRODUCE

Build application gateway using ansible.
Uncomment settings policy_name and policy_type

Change settings on Azure portal. 

Rerun playbook with any ssl_policy settings always results in a App gateway with the default configuration 
ansible-playbook azure_rm_appgateway.yml --ask-vault-pass

```yaml
---
- name: App Gateway
  hosts: localhost

  tasks :

  - name: Manage public ip address for App gateway
    azure_rm_publicipaddress:
      resource_group: myresourcegroup
      name: myappgateway-IP
      allocation_method: dynamic
      state: present

  - name: Manage Application Gateway 
    azure.azcollection.azure_rm_appgateway:
      resource_group: myresourcegroup
      name: myappgateway
      sku:
        name: standard_medium
        tier: standard
        capacity: 2
      gateway_ip_configurations:
        - name: appGatewayIpConfig
          subnet:
            id: /subscriptions/xxxxxxxxx-xxxx-xxxx-xxxxxxxxxxxxxx/resourceGroups/myresourcegroup/providers/Microsoft.Network/virtualNetworks/myvnet/subnets/myvnet-appgateway-subnet
      backend_address_pools:
        - name: backend_pool
          backend_addresses:
            - ip_address: xxx.xxx.xxx.141
            - ip_address: xxx.xxx.xxx.140

        - name: My_Unused_Backend_Pool
          backend_addresses:
            - ip_address: xxx.xxx.0.100
            - ip_address: xxx.xxx.100.100
      backend_http_settings_collection:
        - name: site1.example.com-backend-http-settings
          port: 44300
          protocol: https
          cookie_based_affinity: disabled
          host_name: site1.example.com
          probe: site1.example.com-probe

        - name: site2.example.com-backend-http-settings
          port: 44301
          protocol: https
          cookie_based_affinity: disabled
          host_name: site2.example.com
          probe: site2.example.com-probe
      frontend_ip_configurations:
        - name: appGatewayFrontendIP
          public_ip_address: myappgateway-IP
      #   - subnet:
      #       id: "{{ subnet_id }}"
      frontend_ports:
        - name: port_80
          port: 80
        - name: port_443
          port: 443
      http_listeners:
        - name: site1.example.com-HTTPS
          hostname: site1.example.com
          frontend_ip_configuration: appGatewayFrontendIP
          frontend_port: port_443
          ssl_certificate: TEST_CERT
          protocol: https

        - name: site2.example.com-HTTPS
          hostname: site2.example.com
          frontend_ip_configuration: appGatewayFrontendIP
          frontend_port: port_443
          ssl_certificate: TEST_CERT
          protocol: https
      ssl_policy:
        # policy_name: ssl_policy20170401_s
        # policy_type: predefined
        policy_type: custom
        disabled_ssl_protocols:
         - tls_v1_0
         - tls_v1_1
        cipher_suites: 
          - tls_ecdhe_ecdsa_with_aes_128_gcm_sha256
          - tls_ecdhe_ecdsa_with_aes_256_gcm_sha384  
          - tls_ecdhe_ecdsa_with_aes_128_cbc_sha     
          - tls_ecdhe_ecdsa_with_aes_256_cbc_sha     
          - tls_ecdhe_ecdsa_with_aes_128_cbc_sha256  
          - tls_ecdhe_ecdsa_with_aes_256_cbc_sha384  
          - tls_ecdhe_rsa_with_aes_256_gcm_sha384    
          - tls_ecdhe_rsa_with_aes_128_gcm_sha256    
          - tls_ecdhe_rsa_with_aes_128_cbc_sha       
          - tls_ecdhe_rsa_with_aes_256_cbc_sha       
          - tls_rsa_with_aes_256_gcm_sha384
          - tls_rsa_with_aes_128_gcm_sha256
          - tls_rsa_with_aes_256_cbc_sha256
          - tls_rsa_with_aes_128_cbc_sha256
          - tls_rsa_with_aes_256_cbc_sha
          - tls_rsa_with_aes_128_cbc_sha
        min_protocol_version: tls_v1_2
      request_routing_rules:
        - name: site1.example.com-routing-rule
          rule_type: basic
          backend_address_pool: backend_pool
          backend_http_settings: site1.example.com-routing-backend-http-settings
          http_listener: site1.example.com-routing-HTTPS

        - name: site2.example.com-routing-routing-rule
          rule_type: basic
          backend_address_pool: backend_pool
          backend_http_settings: site2.example.com-routing-backend-http-settings
          http_listener: site2.example.com-routing-HTTPS
      probes:
        - name: site1.example.com-routing-probe
          host: site1.example.com-routing
          interval: 30
          path: '/'
          protocol: https
          timeout: 30
          unhealthy_threshold: 3

        - name: site2.example.com-routing-probe
          host: site2.example.com-routing
          interval: 30
          path: '/'
          protocol: https
          timeout: 30
          unhealthy_threshold: 3
      ssl_certificates:
        - name: TEST_CERT
          data: "MIILmQIBAzCCC18GCSqGSIb3DQEHAaCCC1AEggtMMIILSDCCBf8GCSqGSIb3DQEHBqCCBfAwggXs
                AgEAMIIF5QYJKoZIhvcNAQcBMBwGCiqGSIb3DQEMAQYwDgQI7XZFmwGQJWQCAggAgIIFuDInrsHg
                TlqnDa2wAPdCrQ7Blq28eN3f+uARgTHsPawScgakzzx7S1ZyRnwOjmw4bq4b2ot4nXS9jZAj9dfp
                gMJlLukTHShworKC/gWCYE7SvRSom9WC0tjrHjOy7SNdrMV4WLRbb07735DSJwRaL78s7zesH0W6
                ekPSe+p/xW825YjXfwrZ3kvLBZG0d84cCu9/+t6i/HFmVBZCEY2qRo9l0S07yAmy/V7q4C4G/dlD
                xwACfB7v63BnuuW9rSBDSc63eZra2Z9aXgsEyt0UFh8k+3csqetY35VsqdnKQsAOY5UcP10J+XiZ
                c8flPofS9Jd42B6Vl3LWSgR9NVgqQEi1+40oY6XPhNXz0NJ9Kb93+X1McnxLK534I2uO/8gspq70
                f9/Tq5jU/TJxW0JLvP91Nj0kGQ/QYGrWrxxpESXovRMzUxc+GAQj/GMVGaa5YYDtl2Llz+h+mPrd
                3l3ZknOpTw+5Fvfi1jetle4pX9N1ZmxYqzX+pJl3tOVKo2Qoh5m0TN5LlhjE630M6tqE+MzsMSme
                lWObmjoAcvEQb4lhYTVNU6gvNUQkrnwKIqlKdgb4nsqN/XRIgWRFKLw9XE2G16VzDB45BF4paX0F
                6v6BWvZ3xM1qzzg53C82DS7WNxPud/XFE4Xy7g8ZiOOrZ12iehntFTzwKUIjUVteRlU2upZtBIEQ
                S6QpVR7RP50PKqVDH1cxoWXICdtK/KqKjaGIjZXLtOgRnJ964p8zoEMzUhJAvusTPLr7dYherlQz
                Q9ak1vi7dZm6PwDiQu2uCd0pVIVca8s7Wj0O7gcWleD2tGKfZE1YPBsP4AGCRGQ4irqt4uPECWJU
                //NCAV758pdzWDYDmxXZkbydnhYpXBUgTwYv7CH0dUPpe3rvBPb9RkDMJ3muUTL9o0uFEJoG0cL9
                QMthYyJYzUfdaOm3fobEGaSd3t95jwSNgXDmXhRgtHjhON6AJgDXqHv/4YExQW7aXyOGI9uwnkqp
                COF74HM6dtQBXb4kNy8oGQlPa065bSHYd5JzJ7L6YNVBb83DaA1ZDpDj3uDrfh53EMFoDAOOq5oo
                npWkNkYkunaKvhRUkaL2Jv5TfiQH6/1XgYWikV0YUCNVhszhAmIMCn6dqsT9rgYnOjzk7lj//6uu
                fiRKIZI1UqYgGiwo/r926UmBrAlT9eQa+aqXXDZmXuWN7Qua733A5E7xawNpi9DvYIY2BaPdcK6F
                BTG3AEStpwN43fk7Ga0ubzI05YQed28q/DxtRNyksqKyNxueiD/xxTx1rV9HEAyvpBScB7/m73Te
                WzSZnmNCwwxki+fjoooEmyjVIWqUQMHA8LDXM3wJdCNnntYyalevNow9O+5YS8a/ybmAQCG11Ngf
                HmsxkoPYT3L5HwU4f00Yue93G3VgA+mW21fnrUwY+Rpx5CU0f8sZCxlM1yflGePBoOfyP3ToxZTM
                PVNe3QmDFSuba5I/JIPS5g0Iomm3LBx35kbVVrMweI9aY+pAhOdw2mCqQ69j5qOBedjl7bOPE2vm
                EnmEHhbG0nNUGukZN4JuqMWQfRvpu5DP2ho0q1fMXuKow7NZ8qQuT39eVnETd5mXF4yDmxI4zngq
                sxE9NXVZHtwnGeO86p2cv8bP52QxSzNWXEdK7+MSBdOjbwJAyJ8ykZRPVgbgUaAvzB3Q5oHBuxeF
                sATmhaaU3BblRjZgkbccVFFUFJFetxWd187a3K40AIGzmU9aN6cQaricco6yAu1iGL2rJC++o58T
                GGvj2CJCSQ3xn1NmQQ2WnjvIiSQF/9pQUwgO43ZPPgKI72eoxO7plCx2NLLkG/e+rv/JMsSbwXKu
                R4Yhh/cFPw89FTU4/E8KiCwOw5ZN7rPl9MGz93XmEDugV+lUKcqv+DwkGDiYuet+WNL7qk3yuRgm
                2OOHUn5y1GRcfuAceJ7CdpGlNQTVtN9RscH1TBO5fQ2RZjCCBUEGCSqGSIb3DQEHAaCCBTIEggUu
                MIIFKjCCBSYGCyqGSIb3DQEMCgECoIIE7jCCBOowHAYKKoZIhvcNAQwBAzAOBAjZIK8quYCOSQIC
                CAAEggTIMkTngViRK65nrhz1X8nFOijQFSYC4mG9KfrapIgnIaBWkw4bXHwJmVb2QAs15msecqNf
                sAjQ2NEtXow5UrLIGXrN+zL61GivVZVSql203dCB5HPhP9BPil06bTp6aGDkIurrUfXktLDWoU9E
                1hhyCUgJ60kj4iqBJ4Yn+eqRIwbZ2wQ9JKUGegcql+hiXwZX2KIr8ns4Nd7KuHJQDqjrLyNAtBHW
                kWvdLA17aO4JeHDQ/rbRsF5aNN75GzpEnBLE14EJBbwejqaqIHxiRg/EhHRVZE4cnATFqaas9+J2
                UnkkZl/9BjxVTuNqon84W5Rpmy9zw8pG2r3j9WVnU2x7uck3i4IK7eG0XBhqs7Y/wzemb4J2UX1Z
                s0Eo9W8xSdSBbFvP/tRHlEmExZDzIbJQWtSukFgjtCO41RIKYeFzc0ikl3UgSCVtg82xuPfooLcu
                SUDMDXfaP8qUMkzjMPYeZoL8PJfep6aT/V5TvTp9ahC1O7dth8Xv8ELdR4HNE4i7i67ilsAO1QJP
                lhH64jWAH1jsBDniRZv5ZecfjKVKRNE/jkeuR6Oq8HrC0Cf698hDRKPtUTXdjXs4RRE2aiepU/wm
                XlSf2dxq7CgDmOw+SQuciLSyPeqGiF7MUFTgUojDWvSXehhTVXTKtfw/WtLlvjyPIoMxrfH56Zzj
                xsb3L+nZJofLKzCKxJk3fk04vD6Bl/tXcOyugTkLsgSBzNrDkcMrag/OUB+t1l2KKROCXki8mPw+
                d/t7nWI4iRxQtHw9T/HOqz4Vd4tLqmuDn5GBj+aLbW/vGSAViqXUbEbQfOWxq2c9GaFMJLmMw6Rq
                RH8Ae2lw6w0uLTGhBhLGY4dhRohEaDZVJGD/a2QXlw8xy31achfyS6QT1XrhJHhZkRfrfWiKZ+Fe
                51jTsc00+HKaCv6m6gSQvTD3zkhmmHxExQdq/C98xFl3oc5NDs8JilOOS0vnQioRzecwh5LMgJN2
                0OA9q1E06TyFgbqjOHHTbGdsiW62dGFMHj1GwWl/SeHIl7LJligNohLg7CO13iPN1uY0PFQ3vRYr
                24bIPkO09wvre4DCCCWXKqpY4jPspUNCzpXbnp0bohDUJG0L9UrwEPs3RsMl5isFfPoAUEj7Jfi5
                4lEmFs2Vno/oqVGTe6Lfer82YSRAxHLwCQU+PnVDCWzOT4OFW2s4uHxKOiqbRj7mW4lk5s+ogpmK
                TZkKhOy7S+Bby3qou/KTBcjFPB0b6Y5O1C7V/9vrcECm4C1z9hJYT0SUKkPU4KE8OxOxR1hsgBU+
                NRzSTzGV5pPUDGWBqnPRcP5WExm7lXaTr48tRxN/CIodaeePEIWl47eVyxf5nFnULEB5C4fmbxxz
                1wKrAQpxALuXTBzwtcEKYDW3foZHvv+QLAoCZQeAkKFU3D7NKTIG3aGBNx+c1AwNCM9jW9IVXzJd
                baf9pn72UBVdG/cuNBtjkKE2DMGlqHFCzZP5VQI/o/IQC0MZ4oMGwKYivgk9h/pJCD5W0qDcnBJV
                wAtfcXrn4oScBovpZMrl4zBRL1JCy68MWHFdQgD3xzorVodyCNHeOD2qPj2qq4eXxmmHeGQFKUmk
                J13WW6VFf6RxoXb9GNO4xtO8mjRtGs4k4ggYAoJ+Qc/sMSUwIwYJKoZIhvcNAQkVMRYEFHog2pRf
                WLiVbocoNQx/BOozaFV4MDEwITAJBgUrDgMCGgUABBSuDr4RBO7Rpgv75C0r5JaozxW2ZQQI2HlX
                SJV/FgECAggA"
          password: !vault |
          $ANSIBLE_VAULT;1.1;AES256
          32376530386264636262343736643433393264646565323564393834373831396635353364396432
          3837633138653134313231383862303135613633656639660a626566363230663038313037633433
          63356235353366663439343439616663313935316437633032323363613066346666313562376561
          3730363865323665640a343136316161336136643234383639656239353565396337643363396433
          6436
```

##### EXPECTED RESULTS

Module should apply settings delcared in the play, or ignore making changes to the running configuration of the gateway if they're not declared in the play.

##### ACTUAL RESULTS

As described

```paste below

```

### PR DESCRIPTION
**Pull request by l3ender**
##### SUMMARY
This PR resolves a bug where SSL policy configuration is not set for an application gateway.

Resolves https://github.com/ansible-collections/azure/issues/287.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_appgateway

##### ADDITIONAL INFORMATION

The module input was not being provided to the parameters of the API. I have corrected that while ensuring idempotency, added SSL policy configuration as output in the info module, and added test coverage.

Because the configuration was never applied before, the test cases also needed to be updated because they were using a combination of SSL parameters not supported by the API:

> Error creating the Application Gateway instance: Azure Error: ApplicationGatewaySslPolicyDisabledSslPolicyAndNewSslPropertiesCannotBeSpecifiedTogether
> Message: Properties 'PolicyName', 'PolicyType', 'MinProtocolVersion' and 'CipherSuites' have been introduced for 'SslPolicy' in api-version '2019-06-01'. 'DisabledSslProtocols' cannot be used with the newly introduced properties. Use 'MinProtocolVersion' along with PolicyType 'Custom' instead of 'DisabledSslProtocols' for 'SslPolicy' for Application Gateway 'xxx'.

> Error creating the Application Gateway instance: Azure Error: ApplicationGatewaySslPolicyPropertyNotSupportedForPolicyType
> Message: Specified PolicyType 'Predefined' does not support property 'CipherSuites' for 'SslPolicy' for Application Gateway 'xxx'.

### CODE REVIEW

**Comment by l3ender** [comment]
@Fred-sun Please review. Thank you!

**Comment by l3ender** [comment]
Sorry for the few additional commits after requesting review, but it should be good to go now. Thanks!

**Comment by Fred-sun** [comment]
@l3ender Please!
```e.g.
ERROR: plugins/modules/azure_rm_appgateway_info.py:0:0: return-syntax-error: RETURN.gateways.contains.ssl_policy.description: required key not provided @ data['gateways']['contains']['ssl_policy']['description']. Got None

**Comment by l3ender** [comment]
@Fred-sun - it is now corrected; thank you!

**Comment by l3ender** [comment]
Hi @Fred-sun, anything else needed for this PR? Thank you!

**Comment by Fred-sun** [comment]
@l3ender I will push forward the merger of this PR as soon as possible. Thank you very much!

**Comment by l3ender** [comment]
@Fred-sun I think this PR is good to release. Let me know if I can add anything else--thank you!

File: plugins/modules/azure_rm_appgateway_info.py [hunk 1]
```diff
@@ -192,9 +212,22 @@ def format_response(self, appgw):
             "location": d.get("location"),
             "operational_state": d.get("operational_state"),
             "provisioning_state": d.get("provisioning_state"),
+            "ssl_policy": None if d.get("ssl_policy") is None else {
+                "policy_type": _camel_to_snake(d.get("ssl_policy").get("policy_type", None)),
+                "policy_name": self.ssl_policy_name(d.get("ssl_policy").get("policy_name", None)),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_appgateway_info.py:217]: "ssl_policy" returns the following information without policy_name. Thank you very much!
```e,g,
'ssl_policy': {'policy_type': 'Custom', 'cipher_suites': ['TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256', 'TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256', 'TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384'], 'min_protocol_version': 'TLSv1_1'}

**Comment by xuzhang3** [comment]
LGTM

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**